ql/src/test/queries/clientpositive/udf_array_append.q

```diff
@@ -0,0 +1,44 @@
+--! qt:dataset:src
+
+-- SORT_QUERY_RESULTS
+
+set hive.fetch.task.conversion=more;
+
+DESCRIBE FUNCTION array_append;
+DESCRIBE FUNCTION EXTENDED array_append;
+
+-- evaluates function for array of primitives
+SELECT array_append(array(1, 2, 3, null,3,4), 3);
+
+SELECT array_append(array(1.12, 2.23, 3.34, null,1.11,1.12,2.9),1.12);
+
+SELECT array(1,2,3),array_append(array(1, 2, 3),3);
+
+SELECT array(1,2,3),array_append(array(1, 2, 3),5);
+
+SELECT array_append(array(1, 2, 3), CAST(null AS int));
+
+SELECT array_append(array(1.1234567890, 2.234567890, 3.34567890, null, 3.3456789, 2.234567,1.1234567890),1.1234567890);
+
+SELECT array_append(array(11234567890, 2234567890, 334567890, null, 11234567890, 2234567890, 334567890, null),11234567890);
+
+SELECT array_append(array(array("a","b","c","d"),array("a","b","c","d"),array("a","b","c","d","e"),null,array("e","a","b","c","d")),array("a","b","c","d"));
+
+SELECT array_append(array("aa","bb","cc"),"cc");
```

Comment:
Can we have a failure case, when the first argument is not an array type?

Reply:
Added two negative test cases